pontoon/sync/tests/test_changeset.py

```diff
@@ -300,3 +301,41 @@ def test_obsolete_db(self):
         self.changeset.execute()
         self.main_db_entity.refresh_from_db()
         assert_true(self.main_db_entity.obsolete)
+
+    def test_multiple_locale_authors(self):
+        """
+        Test if authors are allocated per locale properly.
+        Regression tests against bug 1214722.
+
+        Contributors should map to locales following way:
+            first_locale -> first_user, third_user
+            second_locale -> second_user
+        """
+        first_entity, second_entity = EntityFactory.create_batch(2)
+        first_locale, second_locale = LocaleFactory.create_batch(2)
+        first_author, second_author, third_author = UserFactory.create_batch(3)
+
+        TranslationFactory.create(locale=first_locale, user=first_author, entity=first_entity)
+        TranslationFactory.create(locale=second_locale, user=second_author, entity=first_entity)
+        TranslationFactory.create(locale=first_locale, user=third_author, entity=first_entity)
+        # Because of logic contained in Translation.save() we have too update
+        # approved field here.
+        Translation.objects.update(approved=True)
+
```

Comment:
Yeah, the main things you want to be testing for this are:

1. If two translations for different entities are approved and processed, the commit author list should include both
2. If a plural translation includes translations from multiple different authors, all of the authors of the approved translations should be included in the author list.
3. If multiple translations exist for the same entity, only the author of the final approved one should be included.